backend/models/vendor.js

```diff
@@ -3,14 +3,29 @@ const ClientBase = require("./client");
 
 const Schema = mongoose.Schema;
 
+var address = {
+  addressName: { type: String },
+  name: { type: String },
+  addressLine: { type: String },
+  city: { type: String },
+  state: { type: String },
+  zipCode: { type: String },
+  phone: { type: String },
+};
+
+var location = {
```

Comment:
Location should consist of two fields, `longitude` and `latitude`.

Reply:
Changed accordingly.